### ISSUE TITLE
Requirement checks are never called

### ISSUE THREAD

**Issue by zenled** [infinum/iOS-prince-of-versions#38]
It seems that the latest update (`4.0.1`) introduced a new problem. The requirement checks are never called.

In the following example, the requirement check closures are never called, the result is `PrinceOfVersions.PoVError requirementsNotSatisfied`.

```json
{
   "ios":[
      {
         "required_version":"1.2.4",
         "last_version_available":"1.8.0",
         "notify_last_version_frequency":"ALWAYS",
         "requirements":{
            "region":"hr",
            "bluetooth":"5.0"
         }
      }
   ]
}
```

```swift
let url = URL(string: "https://pastebin.com/raw/qtUwSvgR")!

let options = PoVRequestOptions()

options.addRequirement(key: "region"){(value) -> Bool in
   return true // Breakpoint A
}

options.addRequirement(key: "bluetooth"){(value) -> Bool in
   return true // Breakpoint B
}

PrinceOfVersions.checkForUpdates(from: url, options: options){ response in
    switch response.result {
    case .success(let updateResultData):
        print("Update version: \(updateResultData.updateVersion)")
    case .failure:
        print("Failed")
        break
    }
```

I would expect the breakpoints `A` and `B` to be triggered, and the result to be `UpdateStatus.requiredUpdateNeeded`

I did some debugging, and the problem seems to be in the `UpdateInfo.swift` `line:132` the `requirementChecks[$0.key]` returns `nil` (`requirementChecks` is empty) so the requirement is considered as not satisfied.

**Comment by jabou** [comment]
Hi @zenled

Thank you for the report. This should be fixed with the #39 PR. 
Once new update is out, I'll let you know.

**Comment by jabou** [comment]
@zenled I'll close this issue as the new 4.0.2 version is released. If this is still reproducible, feel free to reopen it or open a new one.
